

# Breast Cancer Patient Data Analysis Pipeline

This repository contains a Nextflow pipeline for analyzing breast cancer patient data.

## Software Requirements

- [Nextflow](https://www.nextflow.io/) v20.10.0
- [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) v0.39
- [Salmon](https://combine-lab.github.io/salmon/) v1.5.2
- [R](https://www.r-project.org/) v4.1.0
- [tximport](https://bioconductor.org/packages/release/bioc/html/tximport.html) v1.18.0

## Input Files

The following input files are required to run the pipeline:

- `data/*.fastq.gz`: Paired-end FASTQ files containing the raw sequencing reads.
- `data/Homo_sapiens.GRCh38.102.gtf`: GTF file containing the gene annotations for Homo sapiens.
- `data/design.csv`: CSV file containing the sample IDs and corresponding FASTQ file names.

## Output Files

The pipeline generates the following output files:

- `trmmomatic_output/*_trimmed.fastq.gz`: Trimmed FASTQ files.
- `index_salmon/*`: Salmon index files.
- `salmon_quant_output/*_quant.sf`: Salmon quantification files.
- `merged_quant`: Merged quantification file generated by tximport.

## Running the Pipeline

To run the pipeline, you need to install Nextflow and the required software, and then execute the following command:



nextflow run main.nf


By default, the pipeline runs in the `local` mode, using a single CPU core. To run the pipeline on a cluster, you need to specify a different executor, such as `slurm` or `pbs`.

## Author

sejyotiarna(sejyotiarna@example.com)






